api/models/Images.js

```diff
@@ -55,55 +57,102 @@ module.exports = {
     return imagesWithLabels;
   },
 
-    deleteCascade: async function(id) {
-        let image = await Images.findOne({id: id});
-        if (!image) return;
+  // Helper to add a GPS filters to an existing query object
+  addGpsQueryFilter: function(latMin, latMax, longMin, longMax, query) {
+    if(latMin !== undefined){
+      const latMinF = parseFloat(latMin);
+      if(!Number.isNaN(latMinF)) {
+        query.gpsLatitude = {
+          '>=' : latMinF,
+          '!=' : 999
+        };
+      }
+    }
+    if(latMax !== undefined){
+      const latMaxF = parseFloat(latMax);
+      if(!Number.isNaN(latMaxF)) {
+        if(typeof(query.gpsLatitude) == 'undefined'){
+          query.gpsLatitude = {'!=' : 999};
+        }
+        query.gpsLatitude['<='] = latMaxF;
+      }
+    }
+    if(longMin !== undefined){
+      const longMinF = parseFloat(longMin);
+      if(!Number.isNaN(longMinF)) {
+        query.gpsLongitude = {
+          '>=' : longMinF,
+          '!=' : 999
+        };
+      }
+    }
+    if(longMax !== undefined){
+      const longMaxF = parseFloat(longMax);
+      if(!Number.isNaN(longMaxF)) {
+        if(typeof(query.gpsLongitude) == 'undefined'){
+          query.gpsLongitude = {'!=' : 999};
+        }
+        query.gpsLongitude['<='] = longMaxF;
+      }
+    }
+  },
+
+  hasGpsLocation: function(image) {
+    if(image.gpsLatitude !== 999 && image.gpsLongitude!==999){
+      return true;
+    }
+    return false;
+  },
 
-        // all these things reference Images, so must be also deleted where appropriate
-        console.info('deleting associated objects for imageId=%s', id);
-        await QueuedImages.deleteForImage(id);
-        await Annotations.deleteForImage(id);
-        await GroundTruths.deleteForImage(id);
-        await LineDivisions.deleteForImage(id);
-        await SequencedPairs.deleteForImage(id);
+  deleteCascade: async function(id) {
```

Comment:
Note there are no changes below this point, I just fixed the inconsistent indentation